dnsapi/dns_dnsever.sh

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env sh
+
+#Here is a sample custom api script.
+#This file name is "dns_dnsever.sh"
+#So, here must be a method   dns_dnsever_add()
+#Which will be called by acme.sh to add the txt record to your api system.
+#returns 0 means success, otherwise error.
+#
+#Author: hiska
+#Report Bugs here: https://github.com/hiskang/acme.sh
+#
+########  Public functions #####################
+
+#Usage: dns_myapi_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_dnsever_add() {
+  fulldomain=$1
+  txtvalue=$2
+  _info "Using dnsever"
+  _debug fulldomain "$fulldomain"
+  _debug txtvalue "$txtvalue"
+
+  if [ -z "$DNSEVER_ID" ] || [ -z "$DNSEVER_PW" ]; then
+    DNSEVER_ID=""
+    DNSEVER_PW=""
+    _err "You don't specify dnsever.com ID or PW yet."
+    _err "Please create you key and try again."
+    return 1
+  fi
+
+  #save the api key and email to the account conf file.
+  _saveaccountconf DNSEVER_ID "$DNSEVER_ID"
+  _saveaccountconf DNSEVER_PW "$DNSEVER_PW"
+
+  dnsever_txt "add" "$DNSEVER_ID" "$DNSEVER_PW" "$fulldomain" "$txtvalue"
+  return $?
+}
+
+#Usage: fulldomain txtvalue
+#Remove the txt record after validation.
+dns_dnsever_rm() {
+  fulldomain=$1
+  txtvalue=$2
+  _info "Using dnsever"
+  _debug fulldomain "$fulldomain"
+  _debug txtvalue "$txtvalue"
+
+  if [ -z "$DNSEVER_ID" ] || [ -z "$DNSEVER_PW" ]; then
+    DNSEVER_ID=""
+    DNSEVER_PW=""
+    _err "You don't specify dnsever.com ID or PW yet."
+    _err "Please create you key and try again."
+    return 1
+  fi
+
+  #save the api key and email to the account conf file.
+  _saveaccountconf DNSEVER_ID "$DNSEVER_ID"
+  _saveaccountconf DNSEVER_PW "$DNSEVER_PW"
+
+  dnsever_txt "delete" "$DNSEVER_ID" "$DNSEVER_PW" "$fulldomain" "$txtvalue"
+  return $?
+}
+
+####################  Private functions below ##################################
+
+dnsever_txt() {
+  action="$1"
+  login_id="$2"
+  login_password="$3"
+  fulldomain="$4"
+  txt="$5"
+
+  response=$(_post "login_id=$login_id&login_password=$login_password" "https://kr.dnsever.com/index.html")
+  if [ $? != 0 ] || [ -z "$response" ]; then
+    _err "dnsever_txt:$action ERROR login failed. Please check https://kr.dnsever.com/index.html with login_id=$login_id login_password=$login_password"
+    return 1
+  fi
+
+  _H1="$(grep PHPSESSID "$HTTP_HEADER" | sed s/^Set-//)"
+  export _H1
+
+  response=$(_post "" "https://kr.dnsever.com/start.html")
+  if [ $? != 0 ] || [ -z "$response" ]; then
+    _err "dnsever_txt:$action ERROR login failed. Please check https://kr.dnsever.com/start.html after login"
+    return 1
+  fi
+
+  skey=$(printf "%s\n" "$response" | grep skey | sed -n -e "s/^.*value=['\"]\(.*\)['\"].*/\1/p")
```

Comment:
do not use  `sed -e ` option,  it's not portable.
please consider using  `grep` , `cut` or normal `sed` or `_egrep_o()` function instead.